File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int lowPrice = prices[0];
        int profit = 0;
        
        for(int price : prices){
            lowPrice = Math.min(lowPrice, price);
            profit = Math.max(profit, price - lowPrice);
        }
        
        return profit;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.kt
class Solution {
    fun maxProfit(prices: IntArray): Int {
        var maxProfit = 0;
        var minPrice = prices[0]
        
        for(price in prices){
            minPrice = minPrice.coerceAtMost(price);
            maxProfit = maxProfit.coerceAtLeast(price - minPrice)
        }
        return maxProfit
    }
}